### PR TITLE
Add support for specifying auto generate keys flag

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
To be merged with https://github.com/ballerina-platform/module-ballerina-sql/pull/355

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/1804

## Example
```
Options options = {
  requestGeneratedKeys: ALL
};
Client dbClient = check new (url = url, user = user, password = password, options = options);
sql:ExecutionResult result = checkpanic dbClient->execute("Insert into NumericTypes (int_type) values (20)");
checkpanic dbClient.close();

```

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/java/jdbc/Constants.java [hunk 1]
```diff
@@ -35,11 +35,24 @@
...
+     * Constants for Request Generated Keys field.
+     */
+    public static final class RequestGeneratedKeysValues {
+        public static final String ALL = "ALL";
+        public static final String EXECUTE = "EXECUTE";
+        public static final String BATCH_EXECUTE = "BATCH_EXECUTE";
+    }
+
+
+
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/java/jdbc/Constants.java:57]:
```suggestion
```

**Review by daneshk — DISMISSED** [review]

File: native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java [hunk 2]
```diff
@@ -47,11 +47,16 @@ public static Object createClient(BObject client, BMap<BString, Object> clientCo
...
         BMap properties = null;
         String datasourceName = null;
+        String requestGeneratedKeys = Constants.RequestGeneratedKeysValues.ALL;
         Properties poolProperties = null;
         if (options != null) {
             properties = options.getMapValue(Constants.ClientConfiguration.PROPERTIES);
             BString dataSourceNamVal = options.getStringValue(Constants.ClientConfiguration.DATASOURCE_NAME);
+            BString requestGeneratedKeysVal = options.getStringValue(
+                    Constants.ClientConfiguration.REQUEST_GENERATED_KEYS);
             datasourceName = dataSourceNamVal == null ? null : dataSourceNamVal.getValue();
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java:57]:
```suggestion
            datasourceName = dataSourceNamVal == null ? null : dataSourceNamVal.getValue();
            BString requestGeneratedKeysVal = options.getStringValue(
                    Constants.ClientConfiguration.REQUEST_GENERATED_KEYS);
```

Comment (kaneeldias) [reply native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java]: Fixed in c58ad45

File: native/src/main/java/io/ballerina/stdlib/java/jdbc/Constants.java (hunk 1 above)
Comment (kaneeldias) [reply native/src/main/java/io/ballerina/stdlib/java/jdbc/Constants.java]: Fixed in 35134a1

**Review by niveathika — APPROVED** [review]

File: ballerina/client.bal [hunk 3]
```diff
@@ -124,12 +124,23 @@ public isolated client class Client {
...
-# + properties - the properties of the database which should be applied when getting the connection
+# + properties - The properties of the database which should be applied when getting the connection
+# + requestGeneratedKeys - The database operations for which generated keys should be returned
 public type Options record {|
     string? datasourceName = ();
     map<anydata>? properties = ();
+    Operations requestGeneratedKeys = ALL;
 |};
 
+# Provides a set of database operations.
```
Comment (daneshk) [inline ballerina/client.bal:135]:
```suggestion
# Provides a set of database execute operations.
```

**Comment by niveathika** [comment]
@kaneeldias Lets update the changelog

**Review by daneshk — APPROVED** [review]

File: ballerina/client.bal (hunk 3 above)
Comment (kaneeldias) [reply ballerina/client.bal]: Fixed in 21afee7

**Review by niveathika — APPROVED** [review]

**Pull request merged**